core/src/main/java/org/frankframework/pipes/RegExPipe.java

```diff
@@ -98,15 +101,19 @@ enum RegExFlag {
 	public void configure() throws ConfigurationException {
 		super.configure();
 
-		final int flags = this.flags.isEmpty() ? 0 : Arrays.stream(this.flags.split(","))
-				.map(flag -> RegExFlag.valueOf(flag).getFlag())
+		final int flags = this.flags.isEmpty() ? 0 : this.flags.stream()
+				.map(RegExFlag::getFlag)
 				.reduce(0, (a, b) -> a | b);
```

Comment:
Kan je kort toelichten wat dit magische statement doet?

Reply:
Niks magisch aan deze statement. Iedere enum value wordt naar de correcte flag gemapped. Daarna wordt de lijst gereduced naar een integer met 0 als identity (start) en `(a, b) -> a | b` als functie die twee elementen samenvoegd. De verschillende flags zitten in 1 integer, omdat het een bitmask is.

Reply:
Ik denk dat je de case op `this.flags.isEmpty()` niet nodig hebt?

Reply:
Als je de regex API kent dan is dit statement niet magisch, maar een korte toelichting in een comment helpt degenen die de regex API niet zo goed kennen, of `reduce()` niet kenne. 😄

Reply:
> Niks magisch aan deze statement. Iedere enum value wordt naar de correcte flag gemapped. Daarna wordt de lijst gereduced naar een integer met 0 als identity (start) en `(a, b) -> a | b` als functie die twee elementen samenvoegt. De verschillende flags zitten in 1 integer, omdat het een bitmask is.

Dat klinkt voor mij heel magisch, zelfs met je uitleg. Ik snap de mapping, maar dat reduce statement snap ik niet.

Reply:
Reduce is een methode om een List samen te voegen naar een enkel element. Dit is super handig voor functional programming. Voorbeeld: de som van een lijst kan je dan zo schrijven: `list.reduce(0, (a, b) -> a + b)`. Stel je hebt de lijst [1, 2, 3] dan:
1. 0 + 1 = 1
2. 1 + 2 = 3
3. 3 + 3 = 6
=
6

Dit is hoe de implementatie van reduce er uit ziet:
```java
T result = identity;
for (T element : list)
    result = accumulator.apply(result, element) 

return result
```

In mijn voorbeeld heb ik niet de som nodig, maar een combinatie van alle flags. Als je meerdere flags wilt gebruiken moet je de or operator gebruiken om ze samen te voegen dmv een bitmask. Dus: `2 | 4`. Ik gebruik de or operator om ieder element in de lijst samen te voegen en uiteindelijk op 1 integer uit te komen.